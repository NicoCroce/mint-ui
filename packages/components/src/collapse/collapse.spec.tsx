import React from "react";
import { render } from "@testing-library/react";
import Collapse from "./index";

const Component = () => {
	const [value, setValue] = React.useState(1);
	const PanelContent = () => <div>Panel content</div>;
	return (
		<Collapse
			activeKey={value}
			onChange={(val: number) => setValue(val)}
			panels={[
				{
					panelKey: 1,
					header: {
						text: "Panel Header 1"
					},
					content: <PanelContent />
				},
				{
					panelKey: 2,
					header: {
						text: "Panel Header 2"
					},
					content: <PanelContent />
				},
				{
					panelKey: 3,
					header: {
						text: "Panel Header 3"
					},
					disabled: true,
					content: PanelContent
				}
			]}
		/>
	);
};
test("collapse renders correctly", () => {
	const { asFragment } = render(<Component />);
	expect(asFragment()).toMatchSnapshot();
});
