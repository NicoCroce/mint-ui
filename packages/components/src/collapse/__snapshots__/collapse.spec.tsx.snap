// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapse renders correctly 1`] = `
<DocumentFragment>
  <div
    class="collapse__CollapseWrap-t8cnoc-0 fqDspw"
  >
    <div
      class="tm-panel"
    >
      <div
        class="collapse__StyledHeader-t8cnoc-1 iugnJF tm-panel-header"
      >
        <div
          style="display: inline-flex;"
        >
          <span>
            Panel Header 1
          </span>
        </div>
        <div
          style="display: inline-flex;"
        >
          <i
            class="collapse__StyledEL-t8cnoc-3 hKFcpC"
          >
            <svg
              fill="#a1a6a7"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </i>
        </div>
      </div>
      <div
        class="collapse__Body-t8cnoc-2 boREtH"
      >
        <div>
          Panel content
        </div>
      </div>
    </div>
    <div
      class="tm-panel"
    >
      <div
        class="collapse__StyledHeader-t8cnoc-1 iugnJF tm-panel-header"
      >
        <div
          style="display: inline-flex;"
        >
          <span>
            Panel Header 2
          </span>
        </div>
        <div
          style="display: inline-flex;"
        >
          <i
            class="collapse__StyledEL-t8cnoc-3 hKFcpC"
          >
            <svg
              fill="#a1a6a7"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
    <div
      class="tm-panel"
    >
      <div
        class="collapse__StyledHeader-t8cnoc-1 jjQRyH tm-panel-header"
        disabled=""
      >
        <div
          style="display: inline-flex;"
        >
          <span>
            Panel Header 3
          </span>
        </div>
        <div
          style="display: inline-flex;"
        >
          <i
            class="collapse__StyledEL-t8cnoc-3 hKFcpC"
          >
            <svg
              fill="#a1a6a7"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </i>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
